certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -231,50 +216,49 @@ func (p *DeployableByOlmCheck) installedCSV(operatorData OperatorData, openshift
 	}
 }
 
-func (p *DeployableByOlmCheck) cleanUp(operatorData OperatorData, openshiftengine openshiftengine.OpenshiftEngine) {
+func (p *DeployableByOlmCheck) cleanUp(operatorData OperatorData) {
 
 	log.Debug("Dumping data in artifacts/ directory")
 
-	subs, err := openshiftengine.GetSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
+	subs, err := p.OpenshiftEngine.GetSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
 	if err != nil {
 		log.Error("unable to retrieve the subscription")
 	}
 	p.writeToFile(subs, operatorData.App, "subscription")
 
-	cs, err := openshiftengine.GetCatalogSource(operatorData.App, cli.OpenshiftOptions{Namespace: catalogSourceNS})
+	cs, err := p.OpenshiftEngine.GetCatalogSource(operatorData.App, cli.OpenshiftOptions{Namespace: catalogSourceNS})
 	if err != nil {
 		log.Error("unable to retrieve the catalogsource")
 	}
 	p.writeToFile(cs, operatorData.App, "catalogsource")
 
-	og, err := openshiftengine.GetOperatorGroup(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
+	og, err := p.OpenshiftEngine.GetOperatorGroup(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
 	if err != nil {
 		log.Error("unable to retrieve the operatorgroup")
 	}
 	p.writeToFile(og, operatorData.App, "operatorgroup")
 
-	ns, err := openshiftengine.GetNamespace(operatorData.InstallNamespace)
+	ns, err := p.OpenshiftEngine.GetNamespace(operatorData.InstallNamespace)
 	if err != nil {
 		log.Error("unable to retrieve the namespace")
 	}
 	p.writeToFile(ns, operatorData.InstallNamespace, "namespace")
 
 	log.Trace("Deleting the resources created by Check")
-	openshiftengine.DeleteSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
-	openshiftengine.DeleteCatalogSource(operatorData.App, cli.OpenshiftOptions{Namespace: catalogSourceNS})
-	openshiftengine.DeleteOperatorGroup(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
-	openshiftengine.DeleteNamespace(operatorData.InstallNamespace, cli.OpenshiftOptions{})
+	p.OpenshiftEngine.DeleteSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
```

Comment:
Should all of these be deferred? Is that a best practice?

Reply:
The cleanUp() method (where all these deletions are taking place) is a deferred method. From the code:
```
	err = p.setUp(*operatorData)
	defer p.cleanUp(*operatorData)
```

Reply:
I see. That works.